src/sds.c

```diff
@@ -386,8 +372,13 @@ sds sdsResize(sds s, size_t size, int would_regrow) {
  * 4) The implicit null term.
  */
 size_t sdsAllocSize(sds s) {
-    size_t alloc = sdsalloc(s);
-    return sdsHdrSize(s[-1])+alloc+1;
+    char header = s[-1];
+    /* SDS_TYPE_5 header doesn't contain the size of the allocation */
+    if ((header & SDS_TYPE_MASK) == SDS_TYPE_5) {
+        return zmalloc_size(sdsAllocPtr(s));
```

Comment:
zmalloc_size is not on all platforms! You can see the previous usage of zmalloc_size was inside a jemalloc ifdef

Reply:
If `zmalloc_size` from jemalloc is unavailable, we use our own implementation – https://github.com/valkey-io/valkey/blob/7a9951fb8093c96ceb75da2f7245a1110b614794/src/zmalloc.c#L356

Reply:
I guess I initially thought wasn't ideal because it wasn't accurate, but if we don't have alloc_size on the system we are going to call free anyways.